scico/array.py

```diff
@@ -58,37 +57,22 @@ def ensure_on_device(
                 stacklevel=2,
             )
 
-            arrays[i] = jax.device_put(arrays[i])
         elif not isinstance(
             array,
-            (DeviceArray, scico.blockarray.BlockArray, ShardedDeviceArray),
+            (DeviceArray, BlockArray, ShardedDeviceArray),
         ):
             raise TypeError(
                 "Each item of `arrays` must be ndarray, DeviceArray, BlockArray, or "
                 f"ShardedDeviceArray; Argument {i+1} of {len(arrays)} is {type(arrays[i])}."
             )
 
+        arrays[i] = jax.device_put(arrays[i])
+
     if len(arrays) == 1:
         return arrays[0]
     return arrays
 
 
-def no_nan_divide(
```

Comment:
Where did this go?

Reply:
`scico.numpy.util`. Not all of these guys needed to move, but those that use `BlockArray` seemed to need to.